components/Inputs/Date/DatePicker.vue

```diff
@@ -0,0 +1,199 @@
+    </div>
+</template>
+
+<script>
+import DatePickerHeader from '~/components/Inputs/Date/DatePickerHeader';
+import DatePickerNavigationHeader from '~/components/Inputs/Date/DatePickerNavigationHeader';
+import DatePickerCalendarContent from '~/components/Inputs/Date/DatePickerCalendarContent';
+import {
+    getNextYear,
+    getPreviousYear,
+    getNextMonth,
+    getPreviousMonth,
+    getNextYearsRange,
+    getPreviousYearsRange,
+    getYearsWithinRange,
+    getHeaderForCalendarDaysType,
+    getHeaderForCalendarYearsType,
+    zeroPad,
+    CALENDAR_MONTHS,
+} from '~/model/calendar/calendar';
+import { CalendarType } from '~/model/calendar/CalendarType';
+
+export default {
+    name: 'DatePicker',
+    components: {
+        DatePickerHeader,
+        DatePickerNavigationHeader,
+        DatePickerCalendarContent,
+    },
+    props: {
+        value: {
+            type: Date,
+            default: null,
+        },
+        calendarDate: {
+            type: Date,
+            default() {
+                return new Date();
+            },
+        },
+    },
+    data() {
+        const today = new Date();
+        const year = this.calendarDate.getFullYear();
+        const month = this.calendarDate.getMonth() + 1;
+
+        return {
+            years: getYearsWithinRange([], year),
+            currentYear: today.getFullYear(),
+            currentMonth: today.getMonth() + 1,
+            month,
+            year,
+            selectedCalendarType: CalendarType.DAY,
+            calendarHeader: getHeaderForCalendarDaysType(month, year),
+        };
+    },
+    computed: {
+        header() {
+            if (!this.value) return 'Pick a date';
+
+            const day = this.value.getDate();
+            const month = this.value.getMonth();
+            const year = this.value.getFullYear();
+
+            switch (this.selectedCalendarType) {
+            case CalendarType.DAY:
+                return `${day}.${zeroPad(month, 2)}.${year}`;
+            case CalendarType.MONTH: {
+                const monthDesc = Object.values(CALENDAR_MONTHS)[month];
+                return `${monthDesc} ${year}`;
+            }
+            case CalendarType.YEAR:
+                return year;
+            default:
+                return '';
+            }
+        },
+    },
+    methods: {
+        onDateChange(value) {
+            this.$emit('input', value);
+        },
+        onMonthChange(month) {
+            this.month = month;
+        },
+        onYearChange(year) {
+            this.year = year;
+        },
+        onCalendarTypeChange(type) {
+            this.selectedCalendarType = type;
+        },
+        onCalendarHeaderChange(header) {
+            this.calendarHeader = header;
+        },
+        onPreviousDate() {
+            switch (this.selectedCalendarType) {
+            case CalendarType.DAY: {
+                const { month, year } = getPreviousMonth(this.month, this.year);
```

Comment:
Maybe you should call this variables because they're not readable now (previousMonth, previouseYear).